src/klein/app.py

```diff
@@ -218,20 +237,23 @@ def branch_f(instance, request, *a, **kw):
 
                 self._endpoints[branchKwargs['endpoint']] = branch_f
                 self._url_map.add(
+
                     Rule(
-                        url.rstrip('/') + '/' + '<path:__rest__>',
+                        _urlToString(url)
+                        .rstrip('/') + '/' + '<path:__rest__>',
```

Comment:
OK, so you have a URL object that is smart, but here you turn it until a string and then do string operations on it.

How about: `url.child("<path:__rest__>").asText()` or something?